### PR TITLE
[#3732] refactor: update Python client to align with the Java client API: gravitino_metalake, gravitino_client

### PR DESCRIPTION
**Pull request by shaofengshi**
### What changes were proposed in this pull request?
After the refactoring work in Java client is done (https://github.com/datastrato/gravitino/issues/3626), the Python client should also get updated to align with the Java API.

This PR mainly work on the following python classes:
1. gravitino_metalake.py
2. gravitino_client.py
3. gravitino_admin_client.py

### Why are the changes needed?

To make the API simple and easy to use.

Fix: #3732

### Does this PR introduce _any_ user-facing change?
Change the Python client API.

### How was this patch tested?
The integration tests are updated and passed.

### CODE REVIEW

**Comment by shaofengshi** [comment]
@noidname01 Tim, please review this PR, thanks!

File: clients/client-python/gravitino/client/gravitino_metalake.py [hunk 1]
```diff
@@ -56,40 +54,32 @@ def __init__(self, metalake: MetalakeDTO = None, client: HTTPClient = None):
...
+            A list of the catalog names under this metalake.
         """
-        Namespace.check_catalog(namespace)
-        url = f"api/metalakes/{namespace.level(0)}/catalogs"
+        url = f"api/metalakes/{self.name()}/catalogs"
         response = self.rest_client.get(url)
         entity_list = EntityListResponse.from_json(response.body, infer_missing=True)
         entity_list.validate()
-        return entity_list.identifiers()
+        return [identifier.name for identifier in entity_list.identifiers()]
```
Comment (noidname01) [inline clients/client-python/gravitino/client/gravitino_metalake.py:70]: `name` is a method in `NameIdentifier`, so it should `identifier.name()` here

**Review by noidname01 — CHANGES_REQUESTED** [review]

File: clients/client-python/gravitino/client/gravitino_metalake.py (hunk 1 above)
Comment (shaofengshi) [reply clients/client-python/gravitino/client/gravitino_metalake.py]: solved! and add a test case "test_catalogy.py" to cover it. please review again, thanks!

**Comment by noidname01** [comment]
@shaofengshi LGTM!

**Review by noidname01 — APPROVED** [review]

**Review by xunliu — APPROVED** [review]
LGTM

**Pull request merged**